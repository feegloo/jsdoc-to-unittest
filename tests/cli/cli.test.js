const path = require('path');
const util = require('util');
const fs = require('fs');
const child_process = require('child_process');

const execFile = util.promisify(child_process.execFile);
const writeFileAsync = util.promisify(fs.writeFile);
const unlinkAsync = util.promisify(fs.unlink);

const babelNode = path.resolve(__dirname, '../../node_modules/.bin/babel-node');

const spawn = async (filename, ...args) => {
  const { stdout } = await execFile(babelNode, [
    path.resolve(__dirname, '../../index.js'),
    path.resolve(__dirname, './fixtures/', filename),
    ...args,
  ]);

  return stdout;
};

describe('#cli', () => {
  test('example-file 1 matches snapshot', async () => {
    const stdout = await spawn('example-file1.js');
    expect(stdout).toMatchSnapshot();
  });

  test('jest passes', async () => {
    const name = path.resolve(__dirname, `${Math.random().toString(36).slice(2)}.test.js`);
    const out = await spawn('example-file1.js');
    await writeFileAsync(name, out);
    let passed = true;
    try {
      await execFile(babelNode, [
        path.resolve(__dirname, '../../node_modules/.bin/jest'),
        name,
      ]);
    } catch (ex) {
      passed = ex;
    }

    await unlinkAsync(name);
    expect(passed).toBe(true);
  });
});
