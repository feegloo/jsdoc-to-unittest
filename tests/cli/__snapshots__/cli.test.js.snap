// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#cli example-file 1 matches snapshot 1`] = `
"export function contains(str1, str2) {
  return str1 && str1.indexOf(str2) !== -1;
}

describe('contains', () => {
  test('example 1', () => {
    expect(contains('foo', 'fooo')).toBe(false);
  });

  test('example 2', () => {
    expect(contains('hey', 'ey')).toBe(true);
  });
});
"
`;

exports[`#cli jest passes 1`] = `
"export function contains(str1, str2) {
  return str1 && str1.indexOf(str2) !== -1;
}

describe('contains', () => {
  test('example 1', () => {
    expect(contains('foo', 'fooo')).toBe(false);
  });

  test('example 2', () => {
    expect(contains('hey', 'ey')).toBe(true);
  });
});
"
`;
